### DISCUSSION TITLE
Scripting with neogit?

### DISCUSSION THREAD

**Discussion by gegoune** [NeogitOrg/neogit#1630] [Ideas]
Amazing plugin, only touched the surface what it can do and it seems like you have amazing plans for it! I think I will be able to replace fugitive if I could automate some workflows that I have, namely saving changes to the file, committing those changes and pushing to remote. Is it possible with current state of neogit? I guess neogit would need to expose some functions?

Thanks!

**Comment by RianFuro** [comment]
Just to leave my thoughts on the matter: While I do think we'll expose a stable api at some point to use for scripting, it would most likely involve the use of `async`, so I think we should wait for nvim-lua/plenary.nvim#83 to get merged, transition our code to that and expose an api that way, just so users can have reasonable expectations about the async/await api.

(Also, that way users would need to have plenary, and by extension the async/await set of functions, already installed, so we don't have to worry about that too; and we'd have a place to point to for documentation on the matter :grin:)

**Comment by TimUntersberger** [comment]
@cloggier for now you could look through `neogit.lib`.

Things that might interest you:

* cli.lua
* git/*.lua
